### PR TITLE
[Refactor] Clean up wallet related code

### PR DESCRIPTION
**Pull request by panleone**
## Abstract

This is a continuation of the work started in #197 main changes are:

- From now on `Masterkey` has the only role of internally generating addresses and keys without any context, all external information, for example number of generated addresses, map of known addresses, nAccount used in derivation path has been moved to the `Wallet` class
- Having account info only stored in `Wallet` instead of spread in all files is also a first step toward multi account system.
- I  cleaned up almost all files and removed functions that should be called only internally from the Wallet class: for example `getDerivationPath()`  and `getMasterKey()`
- `Wallet` class has been decoupled from GUI
- `Network` class takes now a `Wallet` object instead of a `Masterkey`

Overall, as you can see from the `git diff`, I think that this is a good clean up and from this point we should finally be able to build a good `Wallet` class that will also manage UTXO, masternode, balance...

---

## Testing
Test all features related to wallet:
- Create a new wallet
- Check that you can send and receive funds
- Check that you can generate new addresses
- Check that the contact system works
- Check that you can encrypt the wallet and change password

If any errors are found, the PR works unexpectedly, or you have viable suggestions to improve the UX or functionality of the PR, let me know!

---
## For future contributions
While the clean up is almost completed there are still a few places that can be improved: for example

- In `bitTrx.js` there is still a call to `wallet.getDerivationPath()`
- the `Masternode` class  should also be refactored and incorporated with the new `Wallet` class

### CODE REVIEW

File: scripts/network.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { cChainParams, COIN } from './chain_params.js';
 import { createAlert } from './misc.js';
 import { Mempool, UTXO } from './mempool.js';
 import { getEventEmitter } from './event_bus.js';
+import { Wallet } from './wallet.js';
```
Comment (Duddino) [inline scripts/network.js:5]:
```suggestion
```
Use jsdoc imports (see below)

File: scripts/network.js [hunk 2]
```diff
@@ -90,15 +91,19 @@ export class HistoricalTx {
 
 /**
  * Virtual class rapresenting any network backend
+ *
  */
 export class Network {
-    constructor(masterKey) {
+    /**
+     * @type {Wallet}
```
Comment (Duddino) [inline scripts/network.js:98]:
```suggestion
     * @type {import('./wallet.js').Wallet}
```

File: scripts/wallet.js [hunk 3]
```diff
@@ -2,6 +2,7 @@ import { parseWIF } from './encoding.js';
 import { generateMnemonic, mnemonicToSeed, validateMnemonic } from 'bip39';
 import { doms, beforeUnloadListener } from './global.js';
 import { getNetwork } from './network.js';
+import { MAX_ACCOUNT_GAP } from './chain_params.js';
 import {
     MasterKey,
```
Comment (Duddino) [inline scripts/wallet.js:7]:
```suggestion
```

Comment (Duddino) [reply scripts/wallet.js:7]: Import as jsdoc comment

**Review by Duddino — CHANGES_REQUESTED** [review]
I like how the new Wallet class works, however there seems to be an exception when you open MPW, other than that I've left some minor nitpicks

**Comment by panleone** [comment]
implemented the review and fixed the problem on opening

**Review by Liquid369 — DISMISSED** [review]
tACK 893afd273ad23969f2239eea28aad1f2c886044b 
Working as per tests

**Review by Duddino — DISMISSED** [review]
tACK

File: scripts/wallet.js [hunk 4]
```diff
@@ -378,7 +476,7 @@ export async function generateWallet(noUI = false) {
         setDisplayForAllWalletOptions('none');
 
         // Update identicon
-        doms.domIdenticon.dataset.jdenticonValue = wallet.getDefaultAddress();
+        doms.domIdenticon.dataset.jdenticonValue = wallet.getAddress();
```
Comment (JSKitty) [inline scripts/wallet.js:479]:
```suggestion
        doms.domIdenticon.dataset.jdenticonValue = await wallet.getAddress();
```

Comment (panleone) [reply scripts/wallet.js]: fixed

**Review by JSKitty — APPROVED** [review]
**tACK** ac66fcd654132f0c829390deb3f7bc29463f91c0

Tested every suggested Test area, tried new wallets, existing wallets, etc, not an issue.

This cleanup has gotten our wallet/key handling code in a much nicer state that I've been wanting for a while, really glad this was handled, on the way to a cleaner and more stable MPW! :rocket:

**Review by Duddino — APPROVED** [review]
tACK I like the new wallet class

**Review by Liquid369 — APPROVED** [review]
tACK ac66fcd654132f0c829390deb3f7bc29463f91c0

**Pull request merged**